src/porepy/models/run_models.py

```diff
@@ -203,16 +201,19 @@ def time_step() -> None:
             solver.solve(model)
         model.after_propagation_loop()
 
-    # Progressbars turned off or tqdm not installed:
-    if not params.get("progressbars", False) or not _IS_TQDM_AVAILABLE:
-        while not model.time_manager.final_time_reached():
-            time_step()
-    # Progressbars turned on:
-    else:
-        # Redirect the root logger, s.t. no logger interferes with with the
-        # progressbars.
-        with logging_redirect_tqdm([logging.root]):
-            # Time loop
+    # Redirect the root logger, s.t. no logger interferes with with the
+    # progressbars.
+    with logging_redirect_tqdm([logging.root]):
+        initial_time_step: float = model.time_manager.dt
+        # Assert that the initial time step is not zero, to avoid division by zero
+        # later on.
+        assert initial_time_step != 0, "Initial time step size must not be zero."
```

Comment:
My sense would be to move this code below the if-else construct, and closer to its actual use.